code/game/machinery/iv_drip.dm

```diff
@@ -1,224 +1,684 @@
+				playsound(src, 'sound/machines/twobeep.ogg', 50)
+				breather.remove_from_mob(breath_mask)
+				breather.update_inv_wear_mask()
+				breath_mask.forceMove(src)
+				breath_mask.canremove = TRUE
+				tank_off()
+				update_icon()
+				return
+			if(tank.air_contents.return_pressure() == 0)
+				src.visible_message(SPAN_WARNING("\The [src] buzzes and automatically closes the valve."))
+				playsound(src, 'sound/machines/buzz-two.ogg', 50)
+				tank_off()
+				update_icon()
+				return
+			if(epp) // Emergency Positive Pressure system forces respiration
+				if(breather.losebreath > 0)
+					if(!epp_active)
+						src.visible_message(SPAN_WARNING("\The [src] flashes a blue light, activating it's Emergency Positive Pressure system!"))
+						playsound(breather, 'sound/machines/windowdoor.ogg', 50)
+						epp_active = TRUE
+						update_icon()
+					tank.distribute_pressure = safe_pressure_min // Constantly adjusts the pressure to keep up with the damage
+					breather.losebreath = 0
+					to_chat(breather, SPAN_NOTICE("You feel fresh air being pushed into your lungs."))
+			update_icon()
 
-	if (istype(W, /obj/item/reagent_containers/blood/ripped))
-		to_chat(user, "You can't use a ripped bloodpack.")
-		return
-	if (istype(W, /obj/item/reagent_containers))
-		if(!isnull(src.beaker))
-			to_chat(user, "There is already a reagent container loaded!")
+	if(istype(attached)) // IV attached
+		if(!beaker)
+			return
+		if(!attached.dna)
+			return
+		if(NOCLONE in attached.mutations)
+			return
+		if(attached.species.flags & NO_BLOOD)
 			return
 
-		user.drop_from_inventory(W,src)
-		src.beaker = W
-		to_chat(user, "You attach \the [W] to \the [src].")
-		src.update_icon()
-		return
-	else
-		return ..()
+		if(!attached.Adjacent(src))
+			iv_rip()
+			update_icon()
+			return
 
+		if(mode) // Injecting
+			if(beaker.reagents.total_volume > 0)
+				beaker.reagents.trans_to_mob(attached, transfer_amount, CHEM_BLOOD)
+				update_icon()
+			if(toggle_stop) // Automatically detaches if the blood volume is at 100%
+				if((beaker.reagents.has_reagent(/decl/reagent/blood) || beaker.reagents.has_reagent(/decl/reagent/saline)) && attached.get_blood_volume() >= 100)
+					visible_message("\The <b>[src]</b> flashes a warning light, disengaging from [attached] automatically!")
+					playsound(src, 'sound/machines/buzz-two.ogg', 100)
+					attached = null
+					blood_message_sent = FALSE
+					update_icon()
+					return
+		else // Taking
+			var/amount = REAGENTS_FREE_SPACE(beaker.reagents)
+			amount = min(amount, transfer_amount)
+
+			if(amount == 0) // If the beaker is full, ping
+				if(prob(5))
+					visible_message("\The <b>[src]</b> pings.")
+					playsound(src, 'sound/machines/ping.ogg', 100)
+				return
+			if(attached.get_blood_volume() < 90 && !blood_message_sent)
+				visible_message(SPAN_WARNING("\The <b>[src]</b> flashes a warning light!"))
```

Comment:
So as I understand it this message only happens once, and is only reset when you toggle? I think this one should repeat every 10s or so because it has the potential to kill people if ignored as it doesn't stop the blood being taken.

Reply:
Yeah, pretty much. I wasn't entirely sure of the original intent so I didn't mess with it, but I'll add a visual overlay as well as the repeating buzz